internal/dispatch/singleflight/singleflight.go

```diff
@@ -82,21 +80,16 @@ func (d *Dispatcher) DispatchCheck(ctx context.Context, req *v1.DispatchCheckReq
 		return d.delegate.DispatchCheck(ctx, req)
 	}
 
-	primary := false
-	v, isShared, err := d.checkGroup.Do(ctx, keyString, func(innerCtx context.Context) (*v1.DispatchCheckResponse, error) {
-		primary = true
+	sharedResp, isShared, err := d.checkGroup.Do(ctx, keyString, func(innerCtx context.Context) (*v1.DispatchCheckResponse, error) {
 		return d.delegate.DispatchCheck(innerCtx, req)
 	})
 
-	singleFlightCount.WithLabelValues("DispatchCheck", strconv.FormatBool(isShared)).Inc()
-	if err != nil {
-		return &v1.DispatchCheckResponse{Metadata: &v1.ResponseMeta{DispatchCount: 1}}, err
+	if sharedResp == nil {
+		sharedResp = &v1.DispatchCheckResponse{Metadata: &v1.ResponseMeta{DispatchCount: 1}}
 	}
 
-	span := trace.SpanFromContext(ctx)
-	singleflighted := isShared && !primary
-	span.SetAttributes(attribute.Bool("singleflight", singleflighted))
```

Comment:
why was this removed?

I understand that technically all requests are single-flighted, but that's not useful information. What we want to know with this is surfacing in the traces when certain traces are _simply not visible_ because they were singleflighted. This means these dispatches don't even appear, making the trace difficult to understand. This sets an attribute that at least lets the user know the reason spans are missing is because it was single-flighted.

Reply:
Turns out it was causing a race condition that the race detector only started to see once I added a test that exercised the code path. I realized we'd likely need a lock to fix it and, IMO, its not worth the overhead simply for a logging property

Reply:
why would it cause a race? unless the same span is being modified concurrently?

Reply:
It was the setting and accessing of the boolean under the handler